### PR TITLE
Refactor isset and session expressions

### PR DESCRIPTION
**Pull request by nimah79**
This PR contains changes to form of these expressions:

+ Expressions like `$_SESSION[$guid]['variableName']` have been replaced with `$gibbon->session->get('variableName')`.
+ Expressions like `isset($_SESSION[$guid]['variableName'])` have been replaced with `$gibbon->session->exists('variableName')`.
+ Expressions containing `isset` in assignmenets have been replaced with null coalescing operator (`??`).

### CODE REVIEW

File: login.php [hunk 1]
```diff
@@ -25,11 +25,11 @@
 setCurrentSchoolYear($guid, $connection2);
 
 //The current/actual school year info, just in case we are working in a different year
-$_SESSION[$guid]['gibbonSchoolYearIDCurrent'] = $_SESSION[$guid]['gibbonSchoolYearID'];
-$_SESSION[$guid]['gibbonSchoolYearNameCurrent'] = $_SESSION[$guid]['gibbonSchoolYearName'];
-$_SESSION[$guid]['gibbonSchoolYearSequenceNumberCurrent'] = $_SESSION[$guid]['gibbonSchoolYearSequenceNumber'];
+$gibbon->session->setet('gibbonSchoolYearIDCurrent', $gibbon->session->get('gibbonSchoolYearID'));
```
Comment (SKuipers) [inline login.php:28]: It looks like these lines ended up formatted incorrectly, perhaps a regex replace gone awry.

File: functions.php [hunk 2]
```diff
@@ -368,14 +368,14 @@ function getPasswordPolicy($guid, $connection2)
 
 function getFastFinder($connection2, $guid)
 {
-    $form = Form::create('fastFinder', $_SESSION[$guid]['absoluteURL'].'/indexFindRedirect.php', 'get');
+    $form = Form::create('fastFinder', $gibbon->session->get('absoluteURL').'/indexFindRedirect.php', 'get');
```
Comment (SKuipers) [inline functions.php:371]: The `$gibbon` object unfortunately isn't available inside these functions. We plan to eventually refactor and replace the functions in this file, but for now the $_SESSION variables will have to left in place.

**Review by SKuipers — CHANGES_REQUESTED** [review]
Hi Nima, thanks for the PR. I've added a couple review comments. Once you push any changes Travis CI will run and test these, if you see a green check mark that's a good indication that they're fixed. Thanks!

File: login.php (hunk 1 above)
Comment (nimah79) [reply login.php]: That's fixed now.

File: functions.php (hunk 2 above)
Comment (nimah79) [reply functions.php]: That's fixed now.

**Comment by nimah79** [comment]
> Hi Nima, thanks for the PR. I've added a couple review comments. Once you push any changes Travis CI will run and test these, if you see a green check mark that's a good indication that they're fixed. Thanks!

As you know, 8 tests are failing in the head of `GibbonEdu:master`. These changes has not affected any other tests.

**Review by SKuipers — APPROVED** [review]
Thanks Nima! I've changed the branch to v21.0.00, which is our current development branch, and resolved a conflict on merge. Also tweaked a variable name in logout.php. All tests are passing and I've reviewed the code, it's looking good. I'm going to merge and test. Thanks! 😃

**Pull request merged**